### ISSUE TITLE
Fetch polyfill failing in Microsoft Edge

### ISSUE THREAD

**Issue by tschaub** [polyfillpolyfill/polyfill-service#1091]
I'm seeing trouble with calls to `fetch` that have `headers` that are arrays of arrays (which looks like if [follows the spec](https://fetch.spec.whatwg.org/#typedefdef-headersinit)).

For example, this request:

```js
fetch('https://example.com', {
  method: 'GET',
  headers: [
    ['accept', 'application/json'],
    ['content-type','application/json']
  ],
  mode: 'cors',
  credentials: 'same-origin'
});
```

is eventually reaching [this code](https://github.com/Financial-Times/polyfill-service/blob/e09871767dcebe671975d3c1c59c1452db5be235/polyfills/fetch/polyfill.js#L383-L385):
```js
request.headers.forEach(function(value, name) {
  xhr.setRequestHeader(name, value);
});
```

Since `headers` is still an array of arrays, this is essentially:
```js
xhr.setRequestHeader(0, ['accept', 'application/json']);
xhr.setRequestHeader(1, ['content-type','application/json']);
```

Calls to `fetch` are coming from `stream-http`, where it looks like https://github.com/jhiesey/stream-http/commit/11d14fc642f4878e9dc8e2329a62e7a8d18895fe might be a relevant change.  I've filed jhiesey/stream-http#71 in hopes of getting more info on that change.  But it looks to me like this might be an issue with the polyfill.

**Comment by tschaub** [comment]
It looks like the polyfill for `Headers` should accept an array [in the constructor](https://github.com/Financial-Times/polyfill-service/blob/e09871767dcebe671975d3c1c59c1452db5be235/polyfills/fetch/polyfill.js#L23-L36).  I've opened github/fetch#485 proposing a fix.

**Comment by triblondon** [comment]
Looks like it was merged but no release yet.  When there's a release we can update our dependency.

**Comment by tschaub** [comment]
@triblondon - the [`whatwg-fetch@2.0.3` release](https://github.com/github/fetch/releases/tag/v2.0.3) includes the fix.  I've opened #1104 to pull it in.  Let me know if this can be updated in another way.  Thanks.

**Comment by JakeChampion** [comment]
This was landed in #1133
